src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/JsonSerializableSyntaxReceiver.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace System.Text.Json.SourceGeneration
+{
+        public class JsonSerializableSyntaxReceiver : ISyntaxReceiver
+        {
+            public List<KeyValuePair<string, IdentifierNameSyntax>> ExternalClassTypes = new List<KeyValuePair<string, IdentifierNameSyntax>>();
```

Comment:
Instead of `KeyValuePair<string, IdentifierNameSyntax>` you can just use tuples:

`List<(string name, IdentifierNameSyntax syntax)>` this has the advantage that you can 'deconstruct' them when you foreach over them

```csharp
foreach ((var name, var syntax) in ExternalClassTypes)
{
  var x = name; //you can now access the two parts of the tuple as independent variables.
  var y = syntax;
}